src/interfaces/token.ts

```diff
@@ -100,6 +105,38 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
         const icon = icons[token.address];
         return icon ? { ...token, icon } : token;
       });
+    } else if (this.chainId === 250) {
+      const metadata = await this.yearn.services.meta.tokens();
+
+      const promises = this.zapTokenAddresses().map(async token => {
+        const price = await this.yearn.services.oracle.getPriceUsdc(token);
+        const metadatum = metadata.find(metadatum => metadatum.address === token);
+
+        let name, symbol, decimals: string;
+
+        if (token === ZeroAddress) {
+          name = "Fantom";
+          symbol = "FTM";
+          decimals = "18";
+        } else {
+          const contract = new Contract(token, TokenAbi, this.ctx.provider.read);
+          // TODO - add multicall and use it for these
+          [name, symbol, decimals] = await Promise.all([contract.name(), contract.symbol(), contract.decimals()]);
```

Comment:
FYI, You can fetch this data using BalancesHelper `tokensMetadata` contract

0x855ffe28019106d089bC018Df18838f8d241c402